### ISSUE TITLE
Letting users bookmark a nearby location for offline use

### ISSUE THREAD

**Issue by maskaravivek** [commons-app/apps-android-commons#1499]
Just letting users bookmark/save a nearby location would be a good idea. Sometimes while browsing I see a few nearby places that I would like to visit and click a picture and then upload the image. Having an option to bookmark such places locally would be great so that users can simply pull out that list and then navigate to that place. The local list could be shown in the same Nearby page(using a toggle or an action bar option) or could be a hamburger item (eg. Saved places)

**Comment by misaochan** [comment]
Thanks for creating a new issue! To respond to your post on the other issue:

> Sorry, I could not understand why #591 needs to be done before implementing this. I was thinking of an offline list that would not be dependent on the APIs. Moreover, the list could contain all items bookmarked by him sorted by the distance.

Sorry, yes, I used the wrong phrasing. I think it would be useful to do #591 first because there isn't really much to bookmark when the user only can see pins within a small radius of where they currently are. However, it's certainly not "necessary" to do #591 first.

I think this is a great idea, either way. :) I personally also think it would be REALLY nice if saved places were all displayed on the map with a "saved" pin that differs from the normal pin, and if they all showed up regardless of where the user was on the map or where they were searching.

**Comment by maskaravivek** [comment]
Yes, showing these places on a map differentiating it from the others sounds like a good idea. :)

**Comment by nicolas-raoul** [comment]
If my understanding is correct, this is now implemented? :-)
Or did I misunderstand something, in particular the offline requirement in particular?

**Comment by misaochan** [comment]
This is a good first implementation but not the final product. :) From the proposal:

> Allow users to bookmark Nearby items that they plan to submit pictures for, and to view all their bookmarked items on the map. In v2.10, users will be able to search for Nearby places anywhere on the map instead of just around their current location. We anticipate that users may find themselves looking at Nearby items that they are interested in taking photos of (e.g. for their photo trip tomorrow), but there is no convenient way for them to save and view all those items on their map. We intend to expand our rudimentary Bookmarks system to allow users to view all their bookmarked places on the Nearby map all the time (regardless of search area), whenever they load it. This feature will also provide the foundation for future enhancements like suggesting itineraries for photo trips. (#1499)

Currently there is only a list of bookmarked items.

**Comment by neslihanturan** [comment]
Current implementation seems good in terms of UI, but after our filtering nearby UI, this UI will need some change (see #1450 ). My suggestion is keeping colors for place states. Adding stars for bookmarks since an item can be has photo/needs photo and bookmarked at the same time, we can't use color anymore.
![path5235](https://user-images.githubusercontent.com/3127881/55962990-a1b0e100-5c7a-11e9-8064-0832fe19bd16.png)

When it comes to filtering, I suggest adding an additional "my bookmarks" item to place types. So that user can see only bookmarks.
![filteringBookmark](https://user-images.githubusercontent.com/3127881/55964937-41bc3980-5c7e-11e9-8c3b-d02d8ac95baa.png)

What do you think?

**Comment by misaochan** [comment]
I love the "my bookmarks" filter! :)

I also like the stars idea, but how about just having an entire star instead of a star inside a pin? IMHO the star inside the pin looks a little strange, and at first glance it is not so easy to tell which ones have stars in them.

**Comment by neslihanturan** [comment]
MY only concern with just stars is, they can be felt something belong the map. Besides I don't want to hide the marker completely because it will cover the marker color so the place type information too. 
Here some other star ideas which does not cover place type information:
![otherStarStyles](https://user-images.githubusercontent.com/3127881/55967970-d83f2980-5c83-11e9-9f9c-8ef9684fc126.png)

**Comment by maskaravivek** [comment]
@neslihanturan Can you share the PNGs for bookmark icons.

**Comment by maskaravivek** [comment]
@misaochan AFAIk as of now we already show the bookmarked places on the map. So all that is left for this task according to the grant proposal is to change the icon for the pin, right? 

> When it comes to filtering, I suggest adding an additional "my bookmarks" item to place types. So that user can see only bookmarks.

I too love the idea for having a filter for bookmarked places.  I guess we can create a separate issue for this task as its different from the original issue.

**Comment by misaochan** [comment]
I think the few things that need doing are:
- Tidy up the bookmark system - it does not work sometimes AFAIK, and was done in bits and pieces. We should have it refactored to MVP, made more robust, unit tests added
- Change the icon for the pin - hopefully we can get other people's opinions on Nes's star icon suggestions :)
- The filter is a core part of this task IMO, there should be a way for the user to view only their bookmarked items and nothing else. @neslihanturan 's filter mockups on this page (bottom 2) are great in my opinion. :)

**Comment by misaochan** [comment]
@nicolas-raoul @VojtechDostal @macgills any thoughts on what sort of icon would look best for displaying bookmarked items on the map? Star, star-within-pin, etc?

**Comment by nicolas-raoul** [comment]
I would stay star-within-pin but I am OK with star too :-)

**Comment by sivaraam** [comment]
To me, a good UI conveys the most essential information well. In that view, I would say it's better if we just focus on choosing an icon that would indicate clearly that the place is bookmarked. For instance, this is how Google maps shows my "starred" places (the pin with the yellow star):

![Screenshot_2020-04-26-00-47-41](https://user-images.githubusercontent.com/12448084/80310720-d1854500-87f9-11ea-9d89-9e2ff3eb2e4f.png)

This is how the pin for that place looked before I bookmarked it:
![Screenshot_2020-04-26-00-47-17](https://user-images.githubusercontent.com/12448084/80310730-d8ac5300-87f9-11ea-8887-c215724a9097.png)

Observe how they didn't try to show any place type information in the bookmarked pin. This looks nice and clean. It would be nice if we do something like that.
